io/src/error.rs

```diff
@@ -103,16 +103,17 @@ impl From<Error> for std::io::Error {
 }
 
 macro_rules! define_errorkind {
-    ($($kind: ident),*) => {
-        #[non_exhaustive]
+    ($($(#[$($attr:tt)*])* $kind:ident),*) => {
         #[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
         /// A minimal subset of [`std::io::ErrorKind`] which is used for [`Error`]. Note that, as with
         /// [`std::io`], only [`Self::Interrupted`] has defined semantics in this crate, all other
         /// variants are provided here only to provide higher-fidelity conversions to and from
         /// [`std::io::Error`].
-        #[allow(missing_docs)]
```

Comment:
In previous commit you added this just to remove it now. Not a big deal but it'd be nicer to just do the commits in the reverse order to decrease the noise in the future.

Reply:
I thought the macro was a big enough change to put it separate.